File: ORB.Core/Models/Metric.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class Metric : ReportingSubEntity
    {
        public Metric()
        {
            MetricUpdates = new HashSet<MetricUpdate>();
        }

        public string MetricCode { get; set; }
        public int DirectorateID { get; set; }
        public int? MeasurementUnitID { get; set; }
        public decimal? TargetPerformanceUpperLimit { get; set; }
        public decimal? TargetPerformanceLowerLimit { get; set; }
        public int? RagOptionID { get; set; }

        public virtual Directorate Directorate { get; set; }
        public virtual MeasurementUnit MeasurementUnit { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual ICollection<MetricUpdate> MetricUpdates { get; set; }
    }
}

File: corporate-reporting/src/types/Metric.ts
import { IDirectorate } from "./Directorate";
import { IUser } from "./User";
import { IEntity } from "./Entity";
import { IReportingEntity, ReportingEntity } from "./ReportingEntity";
import { IMetricUpdate } from "./MetricUpdate";

export interface IMetric extends IReportingEntity {
    MetricCode: string;
    DirectorateID: number;
    Description: string;
    MeasurementUnitID: number;
    TargetPerformanceUpperLimit: number | string;
    TargetPerformanceLowerLimit: number | string;
    LeadUserID: number;
    RagOptionID: number;
    Directorate?: IDirectorate;
    LeadUser?: IUser;
    MeasurementUnit?: IEntity;
    MetricUpdates?: IMetricUpdate[];
}

export class Metric extends ReportingEntity implements IMetric {
    public MetricCode = '';
    public DirectorateID = null;
    public Description = '';
    public ReportingFrequency = null;
    public MeasurementUnitID = null;
    public TargetPerformanceUpperLimit = '';
    public TargetPerformanceLowerLimit = '';
    public LeadUserID = null;
    public RagOptionID = null;
}